src/Metadata/AssemblyMetadata.Constructors.cs

```diff
@@ -43,6 +55,20 @@ private EntityHandle ResolveConstructorReference(ConstructorInfo ctor)
             return methodRef;
         }
 
+        public bool TryGetConstructorDefinitionConstructedType(ConstructorInfo ctor, out MethodBaseDefinitionMetadata metadata) {
+            if (ctor.DeclaringType.IsConstructedGenericType) {
+                var open_constr = ctor.DeclaringType.GetConstructors(AllMethods);
```

Comment:
This whole block is written in a way that is really not idiomatic with C#. You don't need the `goto`. Use Linq instead. You need to clarify explicitly what happen if the target `ctor` is not found (hence remain `null`). Do not propagate the garbage (aka passing the `null` to the next function).